server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletStateChangeIterator.java

```diff
@@ -191,26 +194,29 @@ protected void consume() throws IOException {
       }
 
       // is the table supposed to be online or offline?
-      boolean shouldBeOnline = onlineTables.contains(tls.extent.tableId());
+      final boolean shouldBeOnline = onlineTables.contains(tls.extent.tableId());
+      final boolean isOnDemandTable = onDemandTables.contains(tls.extent.tableId());
+      final boolean onDemandTabletShouldBeHosted = tls.ondemand;
```

Comment:
May be able to compute everything here and then later in the switch stmt only use shouldBeOnline var.

```suggestion
      final boolean shouldBeOnline = onlineTables.contains(tls.extent.tableId()) || (onDemandTables.contains(tls.extent.tableId()) && tls.ondemand);
```

Reply:
those variables are used in the debug log line that follows.